### ISSUE TITLE
Disable lint rule for a table/column

### ISSUE THREAD

**Issue by kunagpal** [skeema/skeema#160]
Hi, thanks for this great project!

Some other linters like [ESLint](https://eslint.org/docs/2.13.1/user-guide/configuring#disabling-rules-with-inline-comments) allow granular lint control. Is there a way to disable a lint rule for a particular table/column? Something like:

```sql
create table `foo` (
  `id` int(11) unsigned NOT NULL
) ENGINE=Archive # ignore: InnoDB engine
```

**Comment by evanelias** [comment]
Thanks for the question. Sorry but this is not supported currently. A few related alternatives:

* If it's fine/expected for some tables to violate a linter rule, be sure to set that rule to `warning` instead of `error`. Warnings are displayed but don't otherwise trigger failure results from `skeema diff`, `skeema push`, or `skeema lint`.

* If you're pushing a change to a table that you know violates a linter rule, you can override a rule on the command-line as needed, for example `skeema push --lint-engine=ignore` (or equivalently `skeema push --skip-lint-engine`)

* You can skip particular tables entirely using an [ignore-table](https://www.skeema.io/docs/options/#ignore-table) regex. This can be used in a .skeema file if you want to omit a particular table from being managed by Skeema entirely. Or it can be used on the command-line ad hoc e.g. `skeema lint --ignore-table='^foo$'`

**Comment by kunagpal** [comment]
@evanelias I see. The use case I'm looking at is lose handling of some rules for legacy tables, with other newer tables completely linted. Would it be possible to have this as a feature request? 😄

**Comment by evanelias** [comment]
This would be a bit complicated, due to how mid-statement SQL comments are not stored by the database server, and then get erased whenever you run `skeema pull`, `skeema format`, or even `skeema lint` unless you specify `--skip-format`... see #72 for explanation.

I can try to think of an alternative approach (not using inline comments) but I'd need to understand the use-case in more detail please: 
* Are you often altering these legacy tables?
* Are there just one or two particular linter rules that these tables violate, or does it vary?
* Are you running Skeema manually or through an automation pipeline?
* Are you linting all tables every time (`skeema lint`) or just altered ones (`skeema diff` / `skeema push`)?

**Comment by kunagpal** [comment]
Thanks for pointing to #72, I now have a clearer picture of why this is complicated. As for the use case:

1. There's no significant difference in alteration frequency across legacy and non-legacy.
2. This is actually quite consistent. Rules like `lint-has-fk` and `lint-charset` are good examples.
3. The latter, although the current setup is bare-minimum: ran into this lint issue whilst exploring 😅 
4. Since this is just an experiment, I'm linting all the tables every time. This can be fine tuned with time!

**Comment by evanelias** [comment]
Makes sense, thanks! Converting this issue to a feature request -- I could see this being handled at the table / file granularity sometime. Column / single-line granularity would be a lot trickier though.

**Comment by morgo** [comment]
This may be useful for redundant indexes that are known to be required. i.e. see the example in https://github.com/skeema/skeema/issues/237
